compactor/src/utils.rs

```diff
@@ -45,6 +45,28 @@ pub struct GroupWithMinTimeAndSize {
 
     /// total size of all file
     pub(crate) total_file_size_bytes: i64,
+
+    /// true if this group was split from a group of many overlapped files
+    pub(crate) overlapped_with_other_groups: bool,
+}
+
+impl GroupWithMinTimeAndSize {
+    /// Make GroupWithMinTimeAndSize for a given set of parquet files
+    pub fn new(files: Vec<ParquetFile>, overlaped: bool) -> Self {
+        let mut group = Self {
+            parquet_files: files,
+            min_time: Timestamp::new(i64::MAX),
+            total_file_size_bytes: 0,
+            overlapped_with_other_groups: overlaped,
+        };
+
```

Comment:
Do you want to make sure this always has at least one file?

```suggestion

assert!(!files.is_empty(), "invalid empty group")
```

Otherwise calling `GroupWithMinTimeAndSize::new(vec![], false)` will be accepted and will result in a `min_time` of `i64::MAX` being observable to the caller.